util/topgen/merge.py

```diff
@@ -32,93 +32,95 @@ def amend_ip(top, ip):
         log.info("TOP doens't use the IP %s. Skip" % ip["name"])
         return
 
-    # Find index of the IP
-    ip_idx = ip_list_in_top.index(ipname)
     # Needed to detect async alert transitions below
     ah_idx = ip_list_in_top.index("alert_handler")
 
-    ip_module = top["module"][ip_idx]
-
-    # Size
-    if "size" not in ip_module:
-        ip_module["size"] = "0x%x" % max(ip["gensize"], 0x1000)
-    elif int(ip_module["size"], 0) < ip["gensize"]:
-        log.error(
-            "given 'size' field in IP %s is smaller than the required space" %
-            ip_module["name"])
+    # Find multiple IPs
+    # Find index of the IP
+    ip_modules = list(
+        filter(lambda module: module["type"] == ipname, top["module"]))
```

Comment:
this is a minor point..i've always wondered since the field is `type`, the variable we compare it against would be better if it were `iptype` instead of `ipname`. I always think instance name when i see the latter.

It's no big deal though.